HVACTemplate clean up initial schedules and schedule type

mjwitte opened the pull request:
Addresses #4860 
Cleaned up the logic in expandobjects for adding the ScheduleTypeLimits, HVACTemplate Any Number; and creating commonly used schedules whenever (and only when) there is an HVACTemplate:\* object present in the idf.

mjwitte committed:
HVACTemplate clean up initial schedules fix 4860

Myoldmopar commented:
@mjwitte I built `ExpandObjects` locally on develop and this branch and compared the outputs from `HVACTemplate-5ZonePurchAir`.  The only diff is that the version on this branch includes an extra `Always 1` schedule that isn't in the original.  If you look at the [CI diffs](http://energyplus.s3-website-us-east-1.amazonaws.com/e7a0cf80563d0206b801d7d48329fc88a14ec92e-d5bf7adba30231eb9b8e300ed8f5a14cf7c99377/HVACTemplate-5ZonePurchAir/x86_64-MacOS-10.9-clang/eplusout.err.diff.html) for the err file, you'll see this added a new warning message because of the new, unused, schedule.  Was this the intent?  When reading the pull request message, it looked like the intent was to not write unused schedules, but maybe there is a side effect.

mjwitte commented:
@Myoldmopar That was not the intent.  I made a bad assumption that Always 1 was always used.  Will redo.

Myoldmopar commented:
Cool, I'll assign myself here so just let me know when you have another revision I can test and merge.

mjwitte committed:
HVACTemplate plan B clean up schedules fix 4860

mjwitte committed:
Correct IDD  - avail sched not required for PTAC, UnitHeater, ZoneEvap

mjwitte commented:
@Myoldmopar 
1. Entirely new fix for the HVACTemplate schedule intitializations.  Moved creation of the schedule type into AddAlwaysSchedule the first time it is called.
2. Also, removed many obsolete IF blocks that inserted "HVACTemplate-Always 1" schedule name when various schedule inputs were left blank.  This is left over from the days when availability schedules were generally a required field.  With current code, most HVACTemplate schedule fields can be passed through, and if they are blank it means always on.
3. Allowing blank schedule names to flow through revealed an IDD error in ZoneHVAC:PackagedTerminalAirConditioner which still has the availability schedule tagged as a required field.  Search of the IDD revealed two other components with the same problem ZoneHVAC:UnitHeater and ZoneHVAC:EvaporativeCooler.  These have been fixed, and the getinput routines were checked to verify that blank is handled properly.

Myoldmopar commented:
Awesome, thanks @mjwitte, let's see what CI says and I can test again if needed.
